### PR TITLE
make maxlag configurable

### PR DESCRIPTION
**Pull request by james-cui**
./refine build

### CODE REVIEW

File: extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java [hunk 1]
```diff
@@ -67,6 +70,14 @@
     private int globalCursor;
     private Map<String, EntityDocument> currentDocs;
     private int batchSize;
+    private static final String MAX_LAG_KEY = "maxLag";
```
Comment (wetneb) [inline extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java]: I would replace `"maxLag"` by something a bit more specific, to make sure there is no collision with anything else: for instance `"wikibase:upload:maxLag"`.

File: extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java [hunk 2]
```diff
@@ -67,6 +70,14 @@
...
+    private static final int MAX_LAG_DEFAULT = 5000;      // 5 second default maxLag
+    private PreferenceStore prefStore = ProjectManager.singleton.getPreferenceStore();
+
+    private int getMaxLag() {
+        return prefStore.get(MAX_LAG_KEY) == null ? MAX_LAG_DEFAULT :
+                Integer.parseInt((String) prefStore.get(MAX_LAG_KEY));
```
Comment (wetneb) [inline extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java]: What happens if `parseInt` fails?  Should it fall back to the default?

File: extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java (hunk 1 above)
Comment (james-cui) [reply extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java]: will improve the naming.

File: extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java (hunk 2 above)
Comment (james-cui) [reply extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java]: good catch, I should wrap the code in a try catch and use default if user put something invalid.

**Comment by james-cui** [comment]
@wetneb, resolved the review comments, do you want to take a look?

**Comment by wetneb** [comment]
Great! Can we have an accompanying test perhaps? Either of your `getMaxLag` function (which could be `protected` instead of `private`), or just by checking that `setMaxLag` is called on the mocked `WikibaseDataEditor`. Both with preferences set and without.

**Comment by james-cui** [comment]
@wetneb  resolved your comments
./refine extensions_test 
passed all tests

**Comment by wetneb** [comment]
Closes #2304.

**Comment by antoine2711** [comment]
@wetneb: here, we use semi columns in the key, but this is not a pattern used elsewhere. Shouldn't dots be used instead?

i.e. `wikibase.upload.maxLag` instead of `wikibase:upload:maxLag`?

Regards, A.

**Comment by tfmorris** [comment]
Yes, we should make this consistent before the release of v3.4.

**Pull request merged**